plugins/kotlin/formatter/src/org/jetbrains/kotlin/idea/formatter/KotlinCommonBlock.kt

```diff
@@ -1067,7 +1067,16 @@ private val INDENT_RULES = arrayOf(
 
     strategy("Block in when entry")
         .within(WHEN_ENTRY)
-        .notForType(BLOCK, WHEN_CONDITION_EXPRESSION, WHEN_CONDITION_IN_RANGE, WHEN_CONDITION_IS_PATTERN, ELSE_KEYWORD, ARROW)
+        .notForType(BLOCK, WHEN_CONDITION_EXPRESSION, WHEN_CONDITION_IN_RANGE, WHEN_CONDITION_IS_PATTERN, ELSE_KEYWORD)
+        .set(Indent.getNormalIndent()),
+
+    strategy("Block after wrapped arrow in when entry")
+        .within(WHEN_ENTRY)
+        .notForType(WHEN_CONDITION_EXPRESSION, WHEN_CONDITION_IN_RANGE, WHEN_CONDITION_IS_PATTERN, ELSE_KEYWORD, ARROW)
+        .forElement { node ->
+            val prevNode = node.prev() ?: return@forElement false
+            prevNode.elementType == ARROW && hasLineBreakBefore(prevNode)
+        }
         .set(Indent.getNormalIndent()),
```

Comment:
Looking at the code example here in ticket: https://youtrack.jetbrains.com/issue/KTIJ-22247/Trailing-comma-adds-wrong-indentation-in-when-statement
I think the root of the issue is here. the type of indentation used.
What is a `Indent.getNormalIndent()`?
Look at the types on indentations used by the closing brace "}" -> https://github.com/JetBrains/intellij-community/blob/master/platform/code-style-api/src/com/intellij/formatting/Indent.java

Maybe `true` can be passed to `getNormalIndent` so that the indent is relative to the parent.

Reply:
Thank you for the hint! I found the flag `enforceIndentToChildren` and looks like is what i want.

Reply:
Happy to help.